### PR TITLE
 Add rate per interval disruption budget policy

### PR DESCRIPTION
**Pull request by tbak**

### CODE REVIEW

File: titus-server-master/src/main/java/com/netflix/titus/master/eviction/service/quota/job/UnhealthyTasksLimitTracker.java [hunk 1]
```diff
@@ -87,23 +87,8 @@ public EvictionQuota getQuota(Reference reference) {
...
-            return Pair.of(
-                    0,
-                    String.format("too few containers started: desired=%s, started=%s", JobFunctions.getJobDesiredSize(job), started)
-            );
-        }
-
-        // We have enough tasks started. Check now how many are healthy.
         int healthy = 0;
-        Map<String, String> unhealthyTasks = new HashMap<>();
+        Map<String, String> notStartedAndHealthyTasks = new HashMap<>();
```
Comment (corindwyer) [inline titus-server-master/src/main/java/com/netflix/titus/master/eviction/service/quota/job/UnhealthyTasksLimitTracker.java]: is this not started and unhealthy tasks or not started and healthy tasks?

File: titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/disruptionbudget/RatePerIntervalDisruptionBudgetRate.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+package com.netflix.titus.api.jobmanager.model.job.disruptionbudget;
+
+import java.time.Duration;
+import java.util.Objects;
+import javax.validation.constraints.Min;
+
+public class RatePerIntervalDisruptionBudgetRate extends DisruptionBudgetRate {
+
+    @Min(1)
```
Comment (corindwyer) [inline titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/disruptionbudget/RatePerIntervalDisruptionBudgetRate.java]: I think there should be maxes on this.

File: titus-server-master/src/main/java/com/netflix/titus/master/eviction/service/quota/job/UnhealthyTasksLimitTracker.java (hunk 1 above)
Comment (tbak) [reply titus-server-master/src/main/java/com/netflix/titus/master/eviction/service/quota/job/UnhealthyTasksLimitTracker.java]: The meaning is not (started and healthy). I agree the naming is confusing. I will make it more explicit.

File: titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/disruptionbudget/RatePerIntervalDisruptionBudgetRate.java (hunk 2 above)
Comment (tbak) [reply titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/disruptionbudget/RatePerIntervalDisruptionBudgetRate.java]: Max makes sense for interval. For `limitPerInterval` it is ok to have it as high as user wants it.
I will set `intervalMs` max to 1 hour.

**Pull request merged**